src/solvers/petsc_dm_wrapper.C

```diff
@@ -80,28 +83,97 @@ namespace libMesh
           LIBMESH_CHKERR(ierr);
 
           // Now set the function pointers for the subDM
+          // Some DMShellGet* functions only exist with PETSc >= 3.12.0.
+
+          // Set Coarsen function pointer
+#if PETSC_VERSION_LESS_THAN(3,12,0)
           if (dm->ops->coarsen)
             {
               ierr = DMShellSetCoarsen(*subdm, dm->ops->coarsen);
               LIBMESH_CHKERR(ierr);
             }
+#else
+          PetscErrorCode (*coarsen)(DM,MPI_Comm,DM*) = nullptr;
```

Comment:
I am actually surprised subDM does not inherit the necessary functions (such as coarsen, refine) from DM.

Reply:
Never mind. I see